poly/helper_test.go

```diff
@@ -0,0 +1,106 @@
+package poly
+
+import (
+	"fmt"
+	"github.com/dedis/crypto/abstract"
+	"github.com/dedis/crypto/config"
+	"github.com/dedis/crypto/random"
+)
+
+var edward = SUITE
+
+func generateKeyPair() *config.KeyPair {
+	keypair := new(config.KeyPair)
+	keypair.Gen(edward, random.Stream)
+	return keypair
+}
+
+func generateKeyPairList(n int) []*config.KeyPair {
+	l := make([]*config.KeyPair, n)
+	for i := 0; i < n; i++ {
+		l[i] = generateKeyPair()
+	}
+	return l
+}
+
+func generatePublicListFromPrivate(private []*config.KeyPair) []abstract.Point {
+	l := make([]abstract.Point, len(private))
+	for i := 0; i < len(private); i++ {
+		l[i] = private[i].Public
+	}
+	return l
+}
+
+// produce M receivers from their private/pub keys
+func generateReceivers(info PolyInfo, keys []*config.KeyPair) []*Receiver {
+	n := len(keys)
+	l := make([]*Receiver, n)
+	for i := 0; i < n; i++ {
+		l[i] = NewReceiver(info, keys[i])
+	}
+	return l
+}
+
+// Produce N dealers with the public keys of the M receivers
+func generateDealers(n int, info PolyInfo, receiverList []abstract.Point) []*Dealer {
+	d := make([]*Dealer, n)
+	for i := 0; i < n; i++ {
+		d[i] = NewDealer(info, generateKeyPair(), generateKeyPair(), receiverList)
+	}
+	return d
+}
+
+// Returns N dealers with M receivers with the right keys / public keys ...
+func generateNDealerMReceiver(info PolyInfo, n, m int) ([]*Dealer, []*Receiver) {
+	receiverKeys := generateKeyPairList(m)
```

Comment:
The first functions are called inside joint_test.go and the latest are used inside schnorr_test.go and are dependant on the first one so... Think of these functions as helpers because both test suite (joint / schnorr) use them.  These functions are separate because they are used independently in each test functions in joint.go. If i do make them united, it will reduce the coverage of my tests greatly. And if i do remove this helper file, I'll just have to write the same functions in both joint.go + schnorr.go.